### PR TITLE
Debugging

### PR DESCRIPTION
**Pull request by pabloslash**
Included feature to perform 1D convolutions of multiple sizes to input sequence. Option to not use any 1D conv and just use the raw input sequence.

### CODE REVIEW

File: RNN.py [hunk 1]
```diff
@@ -13,32 +13,38 @@
 import numpy as np
 from helper import *
 import csv
+import IPython
```
Comment (michaelwiest) [inline RNN.py]: remove this

File: RNN.py [hunk 2]
```diff
@@ -52,14 +58,20 @@ def forward(self, inputs, hidden, predict=False):
...
             outs = [c(inputs)[:, :, self.kernel_size].unsqueeze(-1) for c in self.convs]
+        outs.append(inputs)
+
         c = torch.cat([out for out in outs], 1)
+
         # Turn (batch_size x hidden_size x seq_len) back into (seq_len x batch_size x hidden_size) for RNN
         p = c.transpose(1, 2).transpose(0, 1)
 
+        # IPython.embed()
+
```
Comment (michaelwiest) [inline RNN.py]: get rid of these.

File: RNN.py [hunk 3]
```diff
@@ -13,32 +13,38 @@
...
 
-        p1 = kernel_size
-        self.c1 = nn.Conv1d(input_size, num_filters, kernel_size, padding=p1)
-        dilation = 1
-        p2 = kernel_size + (kernel_size - 1) * dilation
-        self.c2 = nn.Conv1d(input_size, num_filters, kernel_size,
-                            dilation=dilation,
-                            padding=p2)
+        self.convs = []
+        for i in xrange(0,len(kernel_size_l1)):
```
Comment (michaelwiest) [inline RNN.py]: you should also pass an array of dilations. and use those for each of the convolution elements.

File: fasta_sampler.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 from sklearn.preprocessing import LabelEncoder
 from sklearn.preprocessing import OneHotEncoder
 from helper import get_idx
+import IPython
```
Comment (michaelwiest) [inline fasta_sampler.py]: remove

File: main.py [hunk 5]
```diff
@@ -2,11 +2,12 @@
 from RNN import *
 from helper import *
 import csv
+import IPython
 
 batch_size = 20
-kernel_size = 10
+kernel_size = [] # Set different kernel sizes in a list e.g [3, 5, 20]. Empty list will avoid any convolutions.
```
Comment (michaelwiest) [inline main.py]: implement what is said above about dilations.

**Review by michaelwiest — CHANGES_REQUESTED** [review]

**Comment by pabloslash** [comment]
Required changes implemented.

**Pull request merged**